elytron/WFLY-14735-Management-FORM-authentication.adoc

```diff
@@ -0,0 +1,280 @@
+= WFLY-14735 FORM authentication for the management console with MFA
+:author:            Darran Lofthouse
+:email:             darran.lofthouse@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The development of this RFE is delivering a subset of functionality from a much larger planned change,
+this RFE will start with two features:
+
+ * Enabling FORM authentication for admin console access to the application server.
+ * Where custom security realms are in use supporting MFA authentication on the login form to 
+ request additional attributes.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14735[WFLY-14735]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1696[EAP7-1696]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+The main development for this RFE will be a new `elytron-authentication` subsystem 
+in the https://github.com/wildfly/wildfly[WildFly] project.
+
+Changes will also be made within https://github.com/wildfly/wildfly-core[WildFly Core] to
+enable and support the activation of FORM authentication.
+
+Updates will be required to the https://github.com/hal/console[Admin Console] to support
+possibly being required to redirect to the login form and to make use of the logout endpoint.
+
+Security components developed for this RFE will be contained within the
+https://github.com/wildfly-security/wildfly-elytron[WildFly Elytron] project.
+
+=== Other Interested Projects
+
+None
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+==== General Flow
+
+On the HTTP management interface a new context will be made available at the path `/authentication`,
+when a user connects to the root `/` context of the application server they will be redirected to the
+`/authentication` context.  The `/authentication` context will display a login form for the user 
+to authenticate themselves.
+
+The look and feel of the login FORM will make use of `PatternFly` for consistent with the look and
+feel of the web console.
+
+Under the `/authentication` context there will be two sub contexts `/login` and `/logout`, after 
```

Comment:
@jamezp Would you be able to help with some guidance / recommendations here?

Overall this requirement is being delivered in a new subsystem which will be implementing a rest like endpoint. /logout is likely a very simple HTTP request which will result in a cookie being invalidated but /login is the slightly more complex one.

For /login this will receive an authentication request, I think very likely JSON, it will also respons using JSON for an error / additional information state.  The response will also need to be able to set a cookie for the web browser. 

This will not be activate by default to secure the management interface but that is where we are planning to go.  Should we be looking at using RestEasy for these requests or would you recommend something else?